share/dkg/status.go

```diff
@@ -12,7 +13,10 @@ const (
 )
 
 type BitSet map[uint32]bool
-type StatusMatrix map[uint32]BitSet
+type StatusMatrix struct {
```

Comment:
Interesting, this was where I found the issue before when running with `-race` but some basic locking caused deadlock for me

Reply:
Maybe you added locks to both `AllTrue` and `CompleteSuccess`? Because that would deadlock then, I think.